### ISSUE TITLE
aiohttp.ClientConnectorErrors are not retried but should be (?)

### ISSUE THREAD

**Issue by pbourke** [microsoft/sammo#49]
Sometimes an error that occurs during nested generation (ie: a GenerateText component lives in a tree with other components that contain it) ends up becoming part of the tree as an LLMResult. This error escapes `_call_backend` and ends up getting returned as the result of the inference when on_error == "empty_result" in sammo/components.py:122. If on_error is not "empty_result", then the exception is re-raised.

In this case, the error is thrown from aiohttp/connector.py:1087:
```
(<class 'aiohttp.client_exceptions.ClientConnectorError'>, ClientConnectorError(ConnectionKey(host='oar-oai-southcentralus.openai.azure.com', port=443, is_ssl=True, ssl=True, proxy=None, proxy_auth=None, proxy_headers_hash=None), ConnectionResetError(104, 'Connection reset by peer')), <traceback object at 0x7efd97bfb980>)
```

The error is happening in the stack that starts from sammo/runners.py:226 - it happens inside `__aenter__` on session.post.

https://github.com/microsoft/sammo/blob/0fa1dfafd88597e5bf409a3caf1c8ba03e61084e/sammo/runners.py#L223-L237

[this SO issue](https://stackoverflow.com/questions/383738/104-connection-reset-by-peer-socket-error-or-when-does-closing-a-socket-resu) indicates that `ConnectionResetError(104, 'Connection reset by peer'))` is a known transient error that occurs sometimes, perhaps due to GIL timing issues. The suggested fix is to sleep and retry.

A suggested fix for this problem is to wrap the `async with session.post` call in an exception handler that treats `ClientConnectorError` as a `RetriableError` and re-raises anything else. This will allow some time for the transient error condition to pass (looks like a TCP state machine issue).

A potential next step is to introduce a brief sleep before raising the `RetriableError`, which will give time for the TCP connector to become consistent. Ideally the sleep should be handled inside of the Throttler instance, but it seems like it will be common for the Throttler to allow this kind of error to be immediately retried, which would be counter productive.

TL;DR: raise RetriableError on ClientConnectorError because it may represent a transient TCP state issue. Possibly also introduce a brief sleep before the raise.
